src/components/results/securityanalysis/security-analysis-result-n.tsx

```diff
@@ -5,43 +5,110 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-import { FunctionComponent, useMemo } from 'react';
+import { FunctionComponent, useCallback, useMemo } from 'react';
 import {
+    CustomColDef,
     PreContingencyResult,
+    SecurityAnalysisNTableRow,
     SecurityAnalysisResultNProps,
 } from './security-analysis.type';
 import { IntlShape, useIntl } from 'react-intl';
-import { LimitViolation } from './security-analysis.type';
 import { SecurityAnalysisTable } from './security-analysis-table';
 import {
-    computeLoading,
+    FROM_COLUMN_TO_FIELD,
     securityAnalysisTableNColumnsDefinition,
+    securityAnalysisTableNFilterDefinition,
 } from './security-analysis-result-utils';
+import CustomHeaderComponent from 'components/custom-aggrid/custom-aggrid-header';
+import { convertSide } from '../loadflow/load-flow-result-utils';
 
 export const SecurityAnalysisResultN: FunctionComponent<
     SecurityAnalysisResultNProps
-> = ({ result, isLoadingResult }) => {
+> = ({
+    result,
+    isLoadingResult,
+    onSortChanged,
+    sortConfig,
+    updateFilter,
+    filterSelector,
+    filterEnums,
+}) => {
     const intl: IntlShape = useIntl();
 
-    const limitViolations =
-        (result as PreContingencyResult)?.limitViolationsResult
-            ?.limitViolations || [];
-
-    const rows = limitViolations.map((limitViolation: LimitViolation) => {
-        return {
-            subjectId: limitViolation.subjectId,
-            limitType: intl.formatMessage({
-                id: limitViolation.limitType,
-            }),
-            limit: limitViolation.limit,
-            value: limitViolation.value,
-            loading: computeLoading(limitViolation),
-        };
-    });
+    const rows =
+        result?.map((preContingencyResult: PreContingencyResult) => {
+            const { limitViolation } = preContingencyResult;
+            return {
+                subjectId: preContingencyResult.subjectId,
```

Comment:
you can also destruct subjectId from preContingencyResult as you already destructing limitViolation`
```suggestion
                subjectId,
```

Reply:
Done